File: QuasarCode_Library_Maths/Units/ISingleUnit.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace QuasarCode.Library.Maths.Units
{
    public interface ISingleUnit : IUnit, ISymbolUnit
    {
        IQuantity Quantity { get; }
        ISystem System { get; }

        double ConvertToSystemBase(double value, int power = 1);
        double ConvertFromSystemBase(double value, int power = 1);
    }
}

File: QuasarCode_Python/QuasarCode/Maths/Units/_ISingleUnit.py
from QuasarCode.Maths.Units._ISymbolUnit import ISymbolUnit

class ISingleUnit(ISymbolUnit):
    def getQuantity(self):
        raise NotImplementedError("This method MUST be overriden in a child class.")

    def getSystem(self):
        raise NotImplementedError("This method MUST be overriden in a child class.")

    def ConvertToSystemBase(value, power = 1):
        raise NotImplementedError("This method MUST be overriden in a child class.")

    def ConvertFromSystemBase(value, power = 1):
        raise NotImplementedError("This method MUST be overriden in a child class.")